src/accelerate/accelerator.py

```diff
@@ -230,6 +245,51 @@ def prepare_model(self, model):
             model.forward = torch.cuda.amp.autocast()(model.forward)
         return model
 
+    def _prepare_deepspeed(self, *args):
+
+        ds_plugin = self.state.deepspeed_plugin
+        self.ds_config = ds_plugin.ds_config
+
+        batch_size = [obj.batch_size for obj in args if hasattr(obj, "batch_size")]
+        assert len(batch_size) > 0, "You must specify training_dataloader in `accelerate.prepare()` when using DeepSpeed"
+        logger.info("Since you passed both train & eval dataloader, `is_train_batch_min` will decide the `train_batch_size`")
+        batch_size = min(batch_size) if ds_plugin.is_train_batch_min else max(batch_size)
+
+        self.ds_config["train_batch_size"] = batch_size * ds_plugin.gradient_accumulation_steps * self.num_processes
+
```

Comment:
Nothing needs to be done for the `eval_batch_size`?

Reply:
no